Submit stage (Stage 7)

hiaux0 opened the pull request:
## What was done
Figma: https://www.figma.com/file/hTLvqCEzki3WoG5uJbtynw/Prime-Deals?node-id=6159%3A90273
SOT: https://www.notion.so/primedao/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d#461451e1cf8b4f1f9d85b6d4372e4706

#### Styling
- here and there adjusted other components to make the Submit page look as close to the Mocks as possible
- Token details code stolen from @IonelLupu 
- DAO avatar code stolen from @paweljanicki 

## Todo

- [x] e2e test failed in the latest local run
    - when looking into it again, all was fine (maybe through other e2e tests fixed)

## Open (because fnality not in app yet)
SOT: https://www.notion.so/primedao/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d#ac58f0a17f8642edb5409472036d75bc

- [open] Closing this pop up will navigate to the home page. _(open, because mocks don't show cancel, "unofficial mocks" neither)_
- Upon submission
  - [open] go to deal (--> not coded yet that page)
  - [open] when editing a Partnered Deal, all votes must be deleted
- Tweet
  - [open] **User** can tweet the deal or go to deals dashboard
  - [open] Tweet This: We will provide the link to the deals page that is just created


## Testing
- Edit partner deal case
partnered-deal/partnered_deals_stream_hash_3/edit/submit

Just to see a "prettier" submit stage with prefilled data
```ts
    // wizardManager.ts ; line 108
    dealId = "partnered_deals_stream_hash_1";
```

#### Before

#### After

hiaux0 commented on src/resources/elements/popupNotifications/popupNotifications.ts:
```diff
@@ -182,6 +182,7 @@ export class PopupNotifications {
     };
 
     this.queueEventConfig(bannerConfig);
+    console.error(bannerConfig.message);
```
@dkent600 
Thoughts?
Was helpful to get the console log to more quickly find the place in the code

hiaux0 commented on src/resources/elements/primeDesignSystem/ptoggle/ptoggle.scss:
```diff
@@ -72,15 +72,15 @@
       cursor: not-allowed;
       background: transparent;
-      border: 1px solid $Neutral02;
+      border: 1px solid $Secondary05;
```
disabled colors should be purple (not gray)

dkent600 replied:
Should revert this, was a mistake in the figma.  Should be Neutral02

hiaux0 commented on src/resources/elements/primeDesignSystem/ppopup-modal/ppopup-modal.ts:
```diff
@@ -54,4 +54,5 @@ export interface IPopupModalModel {
   buttonTextPrimary?: string;
   buttonTextSecondary?: string;
   confetti?: boolean;
+  className?: string;
```
@dkent600 
Added className config to allow adding className to confirmationPopup

hiaux0 commented on src/wizards/tokenSwapDealWizard/components/stageButtons/stageButtons.ts:
```diff
@@ -1,13 +1,17 @@
 export class stageButtons {
   @bindable wizardManager: any;
-  @bindable showSubmit = false;
+  @bindable showSubmit;
```
Removed the default to better control when to show the submit button.
(Related change: line 28)

hiaux0 commented on src/wizards/tokenSwapDealWizard/stages/submitStage/daoInformation/daoInformation.html:
```diff
@@ -0,0 +1,110 @@
+<template>
```
basically copy pasted from @IonelLupu

dkent600 replied:
would it be reasonable to make this a reusable component, used in both places?  Or are there too many differences?

hiaux0 replied:
At time of implementation, the mocks were not the same.
Now, I asked Jaap, and both places are in sync.

I wanted to avoid too many unrelated changes in this PR.
But would defo want to refactor in a follow up

dkent600 replied:
OK can you please create a ticket for that?  @hiaux0

hiaux0 replied:
#287

hiaux0 commented on src/wizards/tokenSwapDealWizard/stages/submitStage/submitStage.scss:
```diff
@@ -0,0 +1,189 @@
+  color: $Secondary02;
+}
+
+@mixin daoAvatar($size: 32px) {
```
copied from @paweljanicki 

We might want to generalize this in a common place

dkent600 replied:
would it be reasonable to make this a reusable component, used in both places?  Or are there too many differences?

hiaux0 replied:
Only 2 places.

I wanted to avoid too many unrelated changes in this PR.
But would defo want to refactor in a follow up

dkent600 replied:
ok, can you please create a ticket for that?  @hiaux0

hiaux0 replied:
#287

hiaux0 commented on src/services/AlertService.ts:
```diff
@@ -49,7 +49,7 @@ export class AlertService {
 
   public showAlert(config: IAlertModel): Promise<DialogCloseResult> {
-    return this.dialogService.open(Alert, config, { keyboard: true });
+    return this.dialogService.open(Alert, config, { keyboard: true }, config.className);
```
_(General comment about dialog behavior)_

@dkent600

There is this specification in the SOT [1]:
> Closing this pop up will navigate to the home page

I'll need to add an "x" to close the modal.
Do this in this PR or follow up okay?
![grafik](https://user-images.githubusercontent.com/30693990/155839162-5b628a7f-caa4-4e91-9bd8-16dc64af6506.png)


[1]: https://www.notion.so/primedao/Registration-Wizard-74901348d4a443a6891ad16fa1c0b42d#eb8411cc81eb4e66b8b86297f375b89a

dkent600 replied:
Do not add the X.  It was agreed we will not have that in any modal popups.

dkent600 commented on src/wizards/services/WizardService.ts:
```diff
@@ -165,4 +184,10 @@ export class WizardService {
   }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  private deleteVotesForPartneredDeal(wizardManager: WizardManager) {
```
I would like that we avoid "deal" specific stuff in  the WizardService.

hiaux0 replied:
[refac(submit): move congrats modal logic to submitStage](https://github.com/PrimeDAO/prime-deals-dapp/pull/227/commits/eca5c530dd3edb10647f03890b0bed27b2bd13f8)

dkent600 commented on src/wizards/services/WizardService.ts:
```diff
@@ -108,9 +111,25 @@ export class WizardService {
+
+    if (!allStagesValid) return;
+
+    const congratulatePopupModel: IAlertModel = {
```
I would like that we avoid "deal" specific stuff in  the WizardService.  We should find another way to handle a successful click of the submit button (post-validation check).

hiaux0 replied:
Agree.

But due to how the Wizard architecture is implemented, I don't see an "easy" or "quick" fix right now.
So:
1. Spent time thinking of making wizard "deal" unspecific
2. Still do 1. but later

@paweljanicki Do you see a quick win we can implement regarding "making wizard deal unspecific"?

dkent600 replied:
I think putting up the alert modal should be up to the stage, not the wizard.

hiaux0 replied:
Yup I agree.
My question is when this should be refactored?

As I see it, it's the current way of the Wizard architecture

hiaux0 replied:
[refac(submit): move congrats modal logic to submitStage](https://github.com/PrimeDAO/prime-deals-dapp/pull/227/commits/eca5c530dd3edb10647f03890b0bed27b2bd13f8)

dkent600 commented on src/wizards/services/WizardService.ts:
```diff
@@ -4,6 +4,8 @@ import { Router } from "aurelia-router";
 import { STAGE_ROUTE_PARAMETER } from "wizards/tokenSwapDealWizard/dealWizardTypes";
 import { Rule, validateTrigger, ValidationController, ValidationControllerFactory } from "aurelia-validation";
 import { PrimeRenderer } from "resources/elements/primeDesignSystem/validation/primeRenderer";
+import { WizardManager } from "wizards/tokenSwapDealWizard/wizardManager";
```
This must be removed.  The wizardService is meant to be deal-independent.  Further, it would create a circular dependency.

hiaux0 replied:
[refac(wizard): better name for key for wizardStates map](https://github.com/PrimeDAO/prime-deals-dapp/pull/227/commits/e4fcf10ae38cc8df6d5bfec5ce0f89073f3b33e8)

dkent600 reviewed (changes requested):
1) Submit stage should note appear in the steps menu

dkent600 reviewed (changes requested):
I'm still seeing the Submit stage.  Shouldn't be visible in the steps menu.

![image](https://user-images.githubusercontent.com/1821666/157125223-e8766b08-9c9f-4d10-a0b2-578130fd5e8e.png)

dkent600 commented:
Bug:

Isn't showing the social media even though I entered it:

![image](https://user-images.githubusercontent.com/1821666/157130703-f0472a19-2954-4071-8aa1-8fc0fe221245.png)


![image](https://user-images.githubusercontent.com/1821666/157130746-eedb6763-0b2a-4c86-88b8-2c345cf988dd.png)

dkent600 commented:
Either shortened addresses should show the complete address, or should have a tooltip that does.  Note we have ptooltip now.

![image](https://user-images.githubusercontent.com/1821666/157130910-bfabb9ae-bd65-49ad-9159-16eeaccc842b.png)

dkent600 commented:
When optional input is not supplied, indicate such rather than a dash:

This is Launch:

![image](https://user-images.githubusercontent.com/1821666/157131699-3f1e7725-63ca-43a8-9441-fd5806b90671.png)

hiaux0 commented on src/wizards/tokenSwapDealWizard/stages/submitStage/submitStage.scss:
```diff
@@ -0,0 +1,189 @@
+
+$submitCardWidth: 645px;
+
+stage-wrapper[header="Submit"] {
```
bf7ff5ff2d8159ac741102844c2dc2b319d163f7

hiaux0 commented:
- [ ] submit stage should not be in stepper
- [ ] ptoggle color Neutral for disabled
- [ ] social media
- [ ] tooltips
- [ ] optional indicator

hiaux0 commented on cypress/integration/tests/deals/terms.e2e.ts:
```diff
@@ -5,7 +5,7 @@ import {
 import { EditingCard } from "../components/editingCard.e2e";
 
-const UPDATED = "(updated)";
+const UPDATED = "(updated.)";
```
needed for test (some validation in wizard did not pass)

hiaux0 commented on src/wizards/tokenSwapDealWizard/stages/termsStage/termClause/termClause.scss:
```diff
@@ -14,7 +14,7 @@ term-clause {
 
       pre {
-        white-space: pre;
+        white-space: pre-wrap;
```
sidefix: View mode of a clause -> its whitespace misbehaved

dkent600 commented on src/wizards/tokenSwapDealWizard/stages/submitStage/submitStage.scss:
```diff
@@ -0,0 +1,191 @@
+
+  .ppopup-dialog.pcard.gradient {
+    /* Fix pink line appearing */
+    border-right-width: 7px
```
Are you sure this still needs fixing?  Any bug in that popup should be made in the popup, not here.

dkent600 commented on src/services/AlertService.ts:
```diff
@@ -49,7 +49,7 @@ export class AlertService {
 
   public showAlert(config: IAlertModel): Promise<DialogCloseResult> {
-    return this.dialogService.open(Alert, config, { keyboard: true });
+    return this.dialogService.open(Alert, config, { keyboard: true }, config.className);
```
I am concerned that this className is replacing pPopupModal.  This is potentially breaking the system.

I think that `config.className` would best not be passed in this way, rather should continue through in the model to pPopup-Modal where it can be applied at the top of pPopup-Modal.html

hiaux0 replied:
Should we not hardcode the "crucial part for the system" somewhere then?
```ts
className = "pPopupModal", // gets set on ux-dialog-container
```
Ie. move class `pPopupModal` some levels below
![image](https://user-images.githubusercontent.com/30693990/157407367-6a4ff49f-b435-4151-9f93-ea4c79c23ccf.png)

Also, a search for `pPopupModal` is not used anywhere
![image](https://user-images.githubusercontent.com/30693990/157406322-5b58f8d8-7a6f-49f3-af1b-eb7e912720d0.png)

hiaux0 replied:
Currently `pPopupModal` is set in 2 places
![image](https://user-images.githubusercontent.com/30693990/157407729-8224e2eb-7b90-48a0-a1f5-7587113ad449.png)

Imo, the top one should be customizable (as done in this case).
The crucial part is handled in the red circle below

dkent600 commented on src/wizards/tokenSwapDealWizard/components/stageButtons/stageButtons.ts:
```diff
@@ -1,20 +1,45 @@
+    this.wizardService
+      .proceed(this.wizardManager)
+      // eslint-disable-next-line no-console
+      .catch(console.error)
```
our pattern is to use the consoleLogService, not console

dkent600 commented on src/wizards/tokenSwapDealWizard/stages/submitStage/submitStage.html:
```diff
@@ -0,0 +1,134 @@
+        </section>
+
+        <section class="stageSectionContent">
+          <div class="submitContentLabel-sm">Private Execution Period</div>
```
Why "Private Execution Period"?  Should be "On-Chain Execution Period"?

dkent600 reviewed (changes requested):
Font size is too big

![image](https://user-images.githubusercontent.com/1821666/157319631-ee184759-e020-42bf-aedb-574e11d75f7f.png)

hiaux0 commented:
@dkent600 

### font-size in terms section too big
I followed the mocks, what's the expected size?
![image](https://user-images.githubusercontent.com/30693990/157402037-c9fd9752-2bd5-4cfb-a717-3b22649588dc.png)

### popup border fix
Before: 
![image](https://user-images.githubusercontent.com/30693990/157404544-cf3d7420-9224-4a20-9caa-cc7655501ca8.png)
After: moved to `_dialog.scss`

### Text copy
- [ ] awaiting product input

dkent600 reviewed (changes requested):
Changes as we discussed

hiaux0 commented:
### dialog
1. removed glitch fix (only my machine)
2. passed class via bindable to `<ppopup-modal>` (instead of using in the `ux-dialog-container` element)
![grafik](https://user-images.githubusercontent.com/30693990/157501390-89b3100a-afc8-4120-8d46-95ed89c46c91.png)

### New line
Added support for new lines in Proposal.summary and .description, as well as in clauses
![grafik](https://user-images.githubusercontent.com/30693990/157503244-525518cf-40c3-4676-ade8-bd05769f8d93.png)

### Font size in Terms
Reduce from 18 -> 16

hiaux0 commented:
Hmm, only merged development like normal, but dunno why Ionel's commit appear
![grafik](https://user-images.githubusercontent.com/30693990/157633009-bb156334-2eaa-42fa-a6b6-23696c0342bd.png)

dkent600 reviewed (changes requested):
"Connect to Submit" button doesn't do anything.
